src/runtime/pkg/containerd-shim-v2/create.go

```diff
@@ -46,6 +48,45 @@ var defaultStartManagementServerFunc startManagementServerFunc = func(s *service
 	shimLog.Info("management server started")
 }
 
+// calculateSandboxSizing will calculate the number of CPUs and amount of Memory that should
+// be added to the VM if sandbox annotations are provided with this sizing details
+func calculateSandboxSizing(annotations map[string]string) (numCPU, memSizeMB uint32) {
+	// if the annotation isn't defined, or if there's an error in parsing, we'll get a result of zero,
+	// resulting in the numCPU and/or memory being set to zero (ie, noop), so let's just continue silently
+	period, _ := strconv.ParseUint(annotations[ctrAnnotations.SandboxCPUPeriod], 10, 32)
+	quota, _ := strconv.ParseInt(annotations[ctrAnnotations.SandboxCPUQuota], 10, 32)
+	memory, _ := strconv.ParseInt(annotations[ctrAnnotations.SandboxMem], 10, 32)
+
+	return calculateVMResources(period, quota, memory)
+}
+
+// calculatContainerSizing will calculate the number of CPUs and amount of memory that is needed
+// based on the provided LinuxResources
+func calculateContainerSizing(resources *specs.LinuxResources) (numCPU, memSizeMB uint32) {
+	var memory, quota int64
+	var period uint64
+
+	if resources != nil {
+		if resources.CPU != nil && resources.CPU.Quota != nil && resources.CPU.Period != nil {
+			quota = *resources.CPU.Quota
+			period = *resources.CPU.Period
+		}
+
+		if resources.Memory != nil && resources.Memory.Limit != nil {
+			memory = *resources.Memory.Limit
+		}
+
+	}
+
+	return calculateVMResources(period, quota, memory)
```

Comment:
What happens when resources is nil? I guess this is returning `(0,0)`?

Reply:
Yes, that's what it'll do. 

Let me do a quick return at top of function for this scenario - it'll make it more clear, and skip some code in the likely scenario.

Reply:
Done!